Bump timeout for subnet delete from 10 to 20 mins

jen20 opened the pull request:
We've seen subnet deletion failures that we think we can trace to ENIs left behind by NLB. We don't have permission to detach or delete those ENIs, so our only recourse is to wait longer.

Both the state change timeout and the overall resource timeout is increased.

```
$ make testacc TESTARGS='-run=TestAccAWSSubnet'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccAWSSubnet -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccAWSSubnet_importBasic
=== PAUSE TestAccAWSSubnet_importBasic
=== RUN   TestAccAWSSubnet_basic
=== PAUSE TestAccAWSSubnet_basic
=== RUN   TestAccAWSSubnet_ipv6
=== PAUSE TestAccAWSSubnet_ipv6
=== RUN   TestAccAWSSubnet_enableIpv6
=== PAUSE TestAccAWSSubnet_enableIpv6
=== CONT  TestAccAWSSubnet_importBasic
=== CONT  TestAccAWSSubnet_enableIpv6
=== CONT  TestAccAWSSubnet_ipv6
=== CONT  TestAccAWSSubnet_basic
--- PASS: TestAccAWSSubnet_basic (23.14s)
--- PASS: TestAccAWSSubnet_importBasic (23.44s)
--- PASS: TestAccAWSSubnet_enableIpv6 (37.74s)
--- PASS: TestAccAWSSubnet_ipv6 (60.73s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	60.779s
```

bflad commented on aws/resource_aws_subnet.go:
```diff
@@ -320,7 +320,7 @@ func resourceAwsSubnetDelete(d *schema.ResourceData, meta interface{}) error {
 		Pending:    []string{"pending"},
 		Target:     []string{"destroyed"},
-		Timeout:    10 * time.Minute,
+		Timeout:    20 * time.Minute,
```
Should we update this to use `d.Timeout(schema.TimeoutDelete)`? 🤔 I'm a little wary of increasing the `DependencyViolation` timeout for all operators when it really should be more customizable right here.

jen20 replied:
If matching the overall timeout is the new idiom then I'm happy to change this to match?

jen20 commented:
Hi @bflad! Sorry for the belated reply here. I've modified this to use `d.Timeout()` for `Create` and `Delete`, and bumped the delete timeout up to 20 minutes also to solve the actual issue here. I've also rebased this pull request on top of the current `master` branch.

Test output at dde1073:

```shell
$ make testacc TESTARGS='-run=TestAccAWSSubnet'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccAWSSubnet -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccAWSSubnet_importBasic
=== PAUSE TestAccAWSSubnet_importBasic
=== RUN   TestAccAWSSubnet_basic
=== PAUSE TestAccAWSSubnet_basic
=== RUN   TestAccAWSSubnet_ipv6
=== PAUSE TestAccAWSSubnet_ipv6
=== RUN   TestAccAWSSubnet_enableIpv6
=== PAUSE TestAccAWSSubnet_enableIpv6
=== RUN   TestAccAWSSubnet_availabilityZoneId
=== PAUSE TestAccAWSSubnet_availabilityZoneId
=== CONT  TestAccAWSSubnet_importBasic
=== CONT  TestAccAWSSubnet_enableIpv6
=== CONT  TestAccAWSSubnet_basic
=== CONT  TestAccAWSSubnet_availabilityZoneId
=== CONT  TestAccAWSSubnet_ipv6
--- PASS: TestAccAWSSubnet_availabilityZoneId (35.66s)
--- PASS: TestAccAWSSubnet_basic (35.90s)
--- PASS: TestAccAWSSubnet_importBasic (38.01s)
--- PASS: TestAccAWSSubnet_enableIpv6 (58.42s)
--- PASS: TestAccAWSSubnet_ipv6 (96.49s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	96.548s
```

bflad commented:
This has been released in [version 2.14.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2140-june-06-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
